### ISSUE TITLE
How to calculate uncertainty score  at inference stage?

### ISSUE THREAD

**Issue by captainfffsama** [deeplearning-wisc/vos#14]
thanks for your excelent work.
I have read your paper and notice that model will calculate uncertainty score according Eq8 at eval stage, but I can not find it in your code. Forgive me that my work env cannot run your code, so that I study your implement only in github.

Its seems that you just use normal faster-rcnn as eval stage,[(link)](https://github.com/deeplearning-wisc/vos/blob/7126ea0b2c777f1ddab1710032b244902ee4c360/detection/modeling/roihead_gmm.py#L1119)  and I can not find where use the [`all_predicted_prob_vectors`](https://github.com/deeplearning-wisc/vos/blob/7126ea0b2c777f1ddab1710032b244902ee4c360/detection/inference/rcnn_gmm_predictor.py#L50)

**Comment by d12306** [comment]
hi @captainfffsama , thanks for you interest in our work.
In fact, during the inference stage, we run "apply_net.py" which will direct you here.

https://github.com/deeplearning-wisc/vos/blob/7126ea0b2c777f1ddab1710032b244902ee4c360/detection/inference/rcnn_predictor.py#L156

This is where we collect the classification logits and then we store the logits with a .pkl file. After that we calculate the energy score using the "voc_coco_plot.py" in the root directory. The logistic score is just the energy score multiplied by a scalar value in the logistic regression branch. The metrics are essentially the same so you can just use energy score as a metric.

**Comment by FlyCarrot** [comment]
hello @d12306 ,  I just run your code perfectly, and it works well. Thanks for your generous sharing.
But I also have two questions, does the code for equation 8 and equation 9 appear in this code? 
And does the logistic regression branch instance exist in the model structure? Because I think current "voc_coco_plot.py" just use the output of base class head net to calculate the energy.

**Comment by d12306** [comment]
@FlyCarrot ,The logistic score is just the energy score multiplied by a scalar value in the logistic regression branch. The metrics are essentially the same so you can just use energy score as a metric.
